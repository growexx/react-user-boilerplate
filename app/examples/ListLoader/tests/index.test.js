/**
 *
 * Tests for Loader
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import history from 'utils/history';
import request from 'utils/request';
import configureStore from '../../../configureStore';
import Loader from '../index';
let store;
jest.mock('utils/request');
const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <Loader />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );
describe('<Loader />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    request.mockImplementation(() => Promise.resolve({ status: 1 }));
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
});
