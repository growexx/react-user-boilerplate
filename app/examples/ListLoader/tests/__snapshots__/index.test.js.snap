// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render and match the snapshot 1`] = `
.c1 .demo-infinite-container {
  overflow: auto;
  border-radius: 4px;
}

.c1 .demo-loading-container {
  position: absolute;
  bottom: 40px;
  width: 100%;
  text-align: center;
}

.c0 .ant-tabs-tab.ant-tabs-tab-active .ant-tabs-tab-btn {
  color: #4d186e;
}

<div
  class="c0"
>
  <div
    class="ant-tabs ant-tabs-top"
    data-testid="ListTab"
  >
    <div
      class="ant-tabs-nav"
      role="tablist"
    >
      <div
        class="ant-tabs-nav-wrap"
      >
        <div
          class="ant-tabs-nav-list"
          style="transform: translate(0px, 0px);"
        >
          <div
            class="ant-tabs-tab ant-tabs-tab-active"
          >
            <div
              aria-selected="true"
              class="ant-tabs-tab-btn"
              role="tab"
              tabindex="0"
            >
              List With Infinite Loader
            </div>
          </div>
          <div
            class="ant-tabs-tab"
          >
            <div
              aria-selected="false"
              class="ant-tabs-tab-btn"
              role="tab"
              tabindex="0"
            >
              List With Load More
            </div>
          </div>
          <div
            class="ant-tabs-tab"
          >
            <div
              aria-selected="false"
              class="ant-tabs-tab-btn"
              role="tab"
              tabindex="0"
            >
              List With Pagination
            </div>
          </div>
          <div
            class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
          />
        </div>
      </div>
      <div
        class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
      >
        <button
          aria-controls="null-more-popup"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-hidden="true"
          class="ant-tabs-nav-more"
          id="null-more"
          style="visibility: hidden; order: 1;"
          tabindex="-1"
          type="button"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="ant-tabs-content-holder"
    >
      <div
        class="ant-tabs-content ant-tabs-content-top"
      >
        <div
          aria-hidden="false"
          class="ant-tabs-tabpane ant-tabs-tabpane-active"
          role="tabpanel"
          tabindex="0"
        >
          <div
            class="c1"
          >
            <div
              class="demo-infinite-container"
            >
              <div>
                <div
                  class="ant-list ant-list-split"
                >
                  <div
                    class="ant-spin-nested-loading"
                  >
                    <div
                      class="ant-spin-container"
                    >
                      <div
                        class="ant-list-empty-text"
                      >
                        <div
                          class="ant-empty ant-empty-normal"
                        >
                          <div
                            class="ant-empty-image"
                          >
                            <svg
                              class="ant-empty-img-simple"
                              height="41"
                              viewBox="0 0 64 41"
                              width="64"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <g
                                fill="none"
                                fill-rule="evenodd"
                                transform="translate(0 1)"
                              >
                                <ellipse
                                  class="ant-empty-img-simple-ellipse"
                                  cx="32"
                                  cy="33"
                                  rx="32"
                                  ry="7"
                                />
                                <g
                                  class="ant-empty-img-simple-g"
                                  fill-rule="nonzero"
                                >
                                  <path
                                    d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                  />
                                  <path
                                    class="ant-empty-img-simple-path"
                                    d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                  />
                                </g>
                              </g>
                            </svg>
                          </div>
                          <div
                            class="ant-empty-description"
                          >
                            No Data
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="ant-tabs-tabpane"
          role="tabpanel"
          style="display: none;"
          tabindex="-1"
        />
        <div
          aria-hidden="true"
          class="ant-tabs-tabpane"
          role="tabpanel"
          style="display: none;"
          tabindex="-1"
        />
      </div>
    </div>
  </div>
</div>
`;
